feat(cli): support JSON output for install check

betomoedano opened the pull request:
# Why

To improve Doctor compatibility, we have long desired the ability to programmatically read the Expo CLI version check output instead of simply displaying the output from npx expo install --check. This would allow us to reformat the information and potentially

# How

Added new `--json` flag to use with `--check` e.g. `npx expo install --check --json`

# Test Plan

Wrote tests and ran it locally (see screenshots)

<img width="356" alt="image" src="https://github.com/user-attachments/assets/32c93efd-2eed-4951-9699-be40859a374b" />

<img width="362" alt="image" src="https://github.com/user-attachments/assets/adefa063-b9a7-42c8-bf38-0bef02fc4a85" />

<img width="448" alt="image" src="https://github.com/user-attachments/assets/2a6cffc4-d8f2-4ad9-bfc3-f16253ee382e" />

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
-->

- [x] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

betomoedano committed:
Enhance package checking functionality to support JSON output

- Added a `--json` option to the `expo install` command for outputting results in JSON format.
- Updated `checkPackagesAsync` to conditionally log messages and return data in JSON format based on the new option.
- Modified `resolveOptions` to include the new `json` flag in argument parsing.

betomoedano committed:
Update tests to include 'json' flag

betomoedano committed:
Add tests for JSON output in checkPackagesAsync

betomoedano committed:
update changelog

betomoedano committed:
Update JSON output in checkPackagesAsync to include 'upToDate' status

byCedric reviewed (changes requested):
This is looking good! I left 2 comments, one about the Jest spy types and one about the `--json` + `--check` assertion.

Can you also add an E2E test here? https://github.com/expo/expo/blob/6b251273d71373e3248e5ad2912c19131732fb40/packages/%40expo/cli/e2e/__tests__/install-test.ts

betomoedano committed:
Refine JSON output handling in expoInstall command

- Removed the redundant `--json` option description and clarified its usage with the `--check` flag.
- Added validation in resolveOptions to ensure the `--json` flag is only used with `--check`.
- Updated test mocks for console and process exit to use the latest Jest syntax.

betomoedano committed:
Update e2e install test to include the new --json flag

byCedric commented:
You can ignore these CLI test failures, it's our API blocking the request.
